### DISCUSSION TITLE
compiling CCPP-SCM on Derecho

### DISCUSSION THREAD

**Discussion by christianbsch** [NCAR/ccpp-scm#436] [Q&A] [answered]
Hello!,

I am trying to compile CCPP(SCM) on Derecho, previously we used to load a conda environment using:
conda activate /glade/p/ral/jntp/GMTB/CCPP_SCM/conda/ccpp-scm , is this file/directory in another path? 

In addition, there are also some changes, such as the modules: module load hpc-intel/2022.1
module load hpc-mpt/2.25, are those still necessary?

Thank you in advance for any help or comment,

Cheers

Christian

**Comment by hertneky** [comment]
Hi Christian,

For building the 'main' branch on Derecho, you can use the newly added modules:
- module use ccpp-scm/scm/etc/modules
- module load derecho_intel or derecho_gnu
Then you may proceed with the build process as usual. Hope that helps.

Note: There are modulefiles for other platforms on there as well.

**Reply by christianbsch** [reply]
Thank you so much for the quick reply!
I will try again following these instructions...

Regards,

Christian

**Reply by christianbsch** [reply, marked as answer]
Hello,
I just wanted to notify that I have been trying to follow the instructions, however, most of the (sub)routines are trying to load multiple times the module "ncarenv/23.09", which is now ncarenv/23.10, I am trying to update these files/code in a separate directory..
Regards,

C

**Comment by climbfuji** [comment]
ncarenv/23.09 is mandatory for this version of spack-stack used by the scm

**Reply by christianbsch** [reply]
Hello, is still possible to load ncarenv/23.09?, I cant find a way to do it...

**Reply by climbfuji** [reply]
It's the default for me when I log in:
```
JCSDA-L-18146:~ heinzell$ ssh-derecho
Access to and use of this UCAR computer system is limited to authorized use by
UCAR Policies 1-7 and 3-6 and all applicable federal laws, executive orders,
policies and directives. UCAR computer systems are subject to monitoring at all
times to ensure proper functioning of equipment and systems including security
devices, to prevent unauthorized use and violations of statutes and security
regulations, to deter criminal activity, and for other similar purposes.

Users should be aware that information placed in the system is subject to
monitoring and is not subject to any expectation of privacy. Unauthorized use
or abuse will be dealt with according to UCAR Policy, up to and including
criminal or civil penalties as warranted.

By logging in, you are agreeing to these terms.

(heinzell@derecho.hpc.ucar.edu) ncar-two-factor:
Warning: No xauth data; using fake authentication data for X11 forwarding.
Last login: Thu Jan 18 13:44:52 2024 from 128.117.61.142
********************************************************************************
*          Welcome to Derecho - Thursday, March 07 2024
********************************************************************************
           	 Today in the Daily Bulletin (arc.ucar.edu)

       - ATPESC Training Program Deadline Extended

Documentation:        https://ncar-hpc-docs.readthedocs.io
Key module commands:  module list, module avail, module spider, module help
Consulting Services:  https://rchelp.ucar.edu
CISL Help:            cislhelp@ucar.edu -- 303-497-2000
--------------------------------------------------------------------------------
heinzell@derecho5:~> module li

Currently Loaded Modules:
  1) ncarenv/23.09 (S)   3) intel/2023.2.1        5) cray-mpich/8.1.27   7) netcdf/4.9.2
  2) craype/2.7.23       4) ncarcompilers/1.0.0   6) hdf5/1.12.2

  Where:
   S:  Module is Sticky, requires --force to unload or purge



heinzell@derecho5:~>
```

**Reply by christianbsch** [reply]
I will double check it, thank you!

**Reply by mkavulich** [reply]
@christianbsch According to [CISL's update after last month's downtime](https://arc.ucar.edu/articles/609), ncarenv/23.09 recently became the default on Derecho. If you have saved some other version as your default module, you should be able to fix this with the `module restore` command.

**Reply by mkavulich** [reply]
Sorry, the command is `module reset`, not restore

**Reply by christianbsch** [reply]
Problem solved, thank you so much for all the feedback

Cheers,

Christian
